src/launcher/java/org/truffleruby/launcher/CommandLineParser.java

```diff
@@ -435,6 +435,8 @@ private void processArgument() throws CommandLineException {
                     } else if (rubyOpts && argument.equals("--help")) {
                         disallowedInRubyOpts(argument);
                         break FOR;
+                    } else if (argument.equals("--help")) {
+                        config.showHelp = ShowHelp.LONG;
```

Comment:
Should explicitly do this here:
```java
                    // add --help in unknown arguments so that runLauncherAction() actually prints the help
                    config.getUnknownArguments().add(argument);
                    break FOR;
```
because the fallthrough is quite subtle and far.

Reply:
I've done it that way and changed, will return back good point 👍🏼